src/battle.test.c

```diff
@@ -365,18 +365,21 @@ static void test_calculate_armor(CuTest * tc)
     dt.index = 0;
 
     dt.fighter = setup_fighter(&b, du);
-    CuAssertIntEquals_Msg(tc, "default ac", 0, calculate_armor(dt, 0, 0, &magres));
+    CuAssertIntEquals_Msg(tc, "default ac", 0, calculate_armor(dt, 0, 0, select_armor(dt, false), select_armor(dt, true), false));
+    magres = calculate_resistance(dt,
+        select_weapon(dt, false, true) ? select_weapon(dt, false, true)->type : 0,
```

Comment:
Hier wie oben: select_weapon in einem eigenen Test, und z.B. wtype benutzen (Schwert).